src/jquery.multiselect.filter.js

```diff
@@ -181,7 +181,7 @@
       // show/hide optgroups
       $groups.each(function() {
         var $this = $(this);
-        if (!$this.children('li').filter(':visible').length)
+        if (!$this.children('ul').children('li').filter(':visible').length)
```

Comment:
Maybe just use `(!$this.find('li').filter(':visible').length)` since "find" finds descendants vs direct children as "children" does?

Reply:
I was thinking it would be nice to be explicit about the structure I was traversing, but it seems `find` is also generally faster so I'll just go with that.